### DISCUSSION TITLE
i18n support is not compatible with next export. (SSR - NextJS 10)

### DISCUSSION THREAD

**Discussion by lucasReghiniM** [vercel/next.js#20618] [Help]
# i18n support is not compatible with next export.

## NextJS dont run the deploy with i18n

Im using nextJS 10, and the main reason that i choose next 10, is that i can do SSR and use the i18n.
 [Internationalized Routing](https://nextjs.org/docs/advanced-features/i18n-routing
) its a new next js 10 feature and have a page only to tha feature.

But when im gonna do a deploy, this error appears: **i18n support is not compatible with next export.**
Theres nothing about this in internationalized routing page.

### My code
**next.config.js**

```
const withImages = require('next-images')
const path = require('path')

module.exports = withImages({
    esModule: false,
    i18n: {
        locales: ['en-US', 'pt-BR', 'pt-PT', 'es-ES'],
        defaultLocale: 'pt-BR',
      },
});

```

I created a translate archive that make the condition with next router
obs: PT and EN are JSON files with text
```
import * as pt from "./pt";
import * as en from './en';
import { useRouter } from "next/router"

export const traducao = () =>{
  let routes = useRouter();

  let translate;
 
    if (routes.locale == 'pt-PT' || routes.locale == 'pt-BR') {
      translate = pt.default;
    } else {
      translate = en.default;
    }
  
  return translate 
}

```

And the i just use in my project like a function:

`{traducao().homeText.button_text}`

Work well, recognizes the browser language and switch.
Im using deploy script
```
npm run deploy
"deploy": "npm run clean && npm run build && next export -o dist/"
```

### Steps to reproduce

1. Go to 'next.config,js'
2. create the i18n export
3. create a Translate file that recognizes the browser language
4. import JSONs files with your site text
5. Use where you want
6. Try to deploy

## Expected behavior

Its just suppose to work fine and Deploy normal.

## Screenshots

![image](https://user-images.githubusercontent.com/47827740/103380122-aceb1680-4ac6-11eb-9178-e00c28092189.png)
![image](https://user-images.githubusercontent.com/47827740/103380129-b4aabb00-4ac6-11eb-92c7-3bad5ce75ec2.png)
![image](https://user-images.githubusercontent.com/47827740/103380140-bc6a5f80-4ac6-11eb-93b9-41af2f515768.png)
![image](https://user-images.githubusercontent.com/47827740/103380173-d4da7a00-4ac6-11eb-972f-9d6e58c68148.png)

## System information

- OS: Linux Ubuntu
- IDE: VSCode
- Version of Next.js: 10
- Version of Node.js: v15.3.0
- Deployment: next deploy

**Comment by nico-bachner** [comment]
`next export` is not supposed work with Next.js i18n (yet?).

What you can do is deploy your application as a hybrid application (`next build`) for the moment. This should normally still generate static HTML for most your pages at build time, unless you are using `getServerSideProps` or similar

Here's the statement from the [official documentation](https://nextjs.org/docs/advanced-features/i18n-routing):
"Note that Internationalized Routing does not integrate with `next export` as `next export` does not leverage the Next.js routing layer. Hybrid Next.js applications that do not use next export are fully supported."
